test/infrastructure/docker/internal/controllers/dockermachine_controller.go

```diff
@@ -150,8 +150,20 @@ func (r *DockerMachineReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		return ctrl.Result{}, nil
 	}
 
+	if _, hasDeleteAnnotation := machine.Annotations[clusterv1.DeleteMachineAnnotation]; hasDeleteAnnotation {
```

Comment:
I think I can drop it, this was likely before I made `initNodePoolInstanceList` read from the parent Machine.

Reply:
I've since refactored out the `NodePoolInstance` struct and are using the DockerMachines directly, so it's more convenient to just set the label there.